src/resources/js/v2/tickets-modal.js

```diff
@@ -0,0 +1,362 @@
+
+							return;
+						}
+					} );
+				}
+
+				// If we didn't get meta from the API, let's fill with sessionStorage.
+				const local = tribe.tickets.data.getLocal();
+
+				if ( local.meta ) {
+					obj.preFillModalMetaForm( local.meta );
+				}
+
+				// @todo: review this and use the tickets.loader.hide
+				window.setTimeout( obj.loaderHide, 500, tribe.tickets.modal.selectors.loader );
+			}
+		);
+
+		tribe.tickets.loader.hide( $document );
+	};
+
+	/**
+	 * Attempts to hydrate a dynamically-created attendee form "block" from sessionStorage data.
+	 *
+	 * @since TBD
+	 *
+	 * @param {number} length The "skip" index.
+	 */
+	obj.maybeHydrateAttendeeBlockFromLocal = function( length ) {
+		$.when(
+			tribe.tickets.data.getData()
+		).then(
+			function( data ) {
+				if ( ! data.meta ) {
+					return;
+				}
+
+				const cartSkip = data.meta.length;
+				if ( length < cartSkip ) {
+					obj.modal.preFillModalMetaForm( data.meta );
+					return;
+				}
+				const $attendeeForm = $( obj.selectors.metaForm );
+				const $newBlocks = $attendeeForm.find( obj.selectors.metaItem ).slice( length - 1 );
+
+				if ( ! $newBlocks ) {
+					return;
+				}
+
+				$newBlocks.find( obj.selectors.metaField ).each(
+					function() {
+						const $this = $( this );
+						const name = $this.attr( 'name' );
+						const storedVal = data[ name ];
+
+						if ( storedVal ) {
+							$this.val( storedVal );
+						}
+					}
+				);
+			}
+		);
+	};
+
+	/**
+	 * Handles the initialization of the scripts when Document is ready.
+	 *
+	 * @since TBD
+	 *
+	 * @return {void}
+	 */
+	obj.ready = function() {
+		// Silence is golden.
+
+		/**
+		 * Handles storing data to local storage
+		 */
+		$( tribe.dialogs.events ).on(
+			'tribe_dialog_close_ar_modal', // @todo: see how to do something more container based.
+			function() {
+				tribe.tickets.data.storeLocal();
+			}
+		);
+
+		/**
+		 * When "Get Tickets" is clicked, update the modal.
+		 *
+		 * @since TBD
+		 */
+		$( tribe.dialogs.events ).on(
+			'tribe_dialog_show_ar_modal',
+			function() {
+
+				console.log( $( this ) );
+
+				const $form = $( obj.selectors.form );
+				const $modalCart = $( obj.selectors.cartForm );
+				// @todo: Fix how this is handling stuff (use container based).
+
+				const $tribeTicket = $( tribe.tickets.block.selectors.container );
+
+				const $cartItems = $tribeTicket.find( tribe.tickets.block.selectors.item );
+
+				tribe.tickets.loader.show( $form );
+
+				$cartItems.each(
+					function() {
+						const $blockCartItem = $( this );
+						const id = $blockCartItem.data( 'ticketId' );
+						const $modalCartItem = $modalCart.find( '[data-ticket-id="' + id + '"]' );
+
+						if ( ! $modalCartItem ) {
+							return;
+						}
+
+						tribe.tickets.block.updateItem( id, $modalCartItem, $blockCartItem );
+					}
+				);
+
+				obj.initModalFormPreFills();
+
+				tribe.tickets.block.updateFormTotals( $modalCart );
+
+				tribe.tickets.loader.hide( $form );
+			}
```

Comment:
Let's make it into a separate method.

Reply:
This is not ready for review, pushed so Scott could know where we were at.